### PR TITLE
[PIM-6517] Move filters

### PR DESCRIPTION
**Pull request by pierallard**
### Remaining

- [x] Display manage filters
- [x] Ensure the "old manage filters" works
- [x] Price filter
- [x] DCI
- [x] Rollback the category column on click
- [x] Bug on attributes page (position updated)
- [x] Add "done" button
- [x] Product scope switcher is not correclty displayed
- [x] Align dropdown to the left on the criteria selectors

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

**Comment by tamarasaurus** [comment]
Behats to unskip (offscreen filters issue)

https://github.com/akeneo/pim-enterprise-dev/pull/3085/files#diff-c74f219240102926fc90ea04bea3c772R111

https://github.com/akeneo/pim-community-dev/pull/6700/files#diff-96c78797334b0114adc29ab4b292f95eR37

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js [hunk 1]
```diff
@@ -123,19 +125,17 @@ function($, _, mediator) {
...
-                    var length = htmlContent ? htmlContent.length : 0;
-
-                    return length;
-                });
-
-                this.minimumWidth = $(longestElement).find('span:first').width() + margin;
+                const margin = 100;
+                const elements = this.getWidget().find('.ui-multiselect-checkboxes li');
+                const longest = _.max(_.map(elements, function (element) {
+                    console.log($(element).find('span:first').width());
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js]: remove

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js [hunk 2]
```diff
@@ -123,19 +125,17 @@ function($, _, mediator) {
...
+                    return $(element).find('span:first').width();
+                }));
+
+                this.minimumWidth = longest + margin;
             }
 
+            console.log(this.minimumWidth);
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js]: remove

File: features/Context/Page/Base/Grid.php [hunk 3]
```diff
@@ -730,6 +730,7 @@ protected function clickOnFilterToManage($filterName)
 
             if (null !== $filterElement && $filterElement->isVisible()) {
                 $filterElement->click();
+                $manageFilters->find('css', '.done-button')->click();
```
Comment (pierallard) [inline features/Context/Page/Base/Grid.php]: .close

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/abstract-filter.js [hunk 4]
```diff
@@ -438,6 +460,38 @@ function($, _, Backbone, app) {
...
+            }
+
+            let expectedTop = this.$el.offset().top;
+            if ((expectedTop + criteria.outerHeight() > body.height()) &&
+                expectedTop + this.$el.height() - criteria.outerHeight() > 0) {
+                criteria.css({ top: expectedTop + this.$el.height() - criteria.outerHeight() });
+            } else {
+                criteria.css({ top: expectedTop });
+            }
+        },
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/abstract-filter.js]: reagir au scroll

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/choice-filter.js [hunk 5]
```diff
@@ -146,17 +114,38 @@ function($, _, __, app, TextFilter, initSelect2) {
...
+                .removeClass('AknDropdown-menuLink--active')
+                .removeClass('active');
+
+            const currentOperatorChoice = this.$el.find('.operator .operator_choice[data-value="' + operator + '"]');
+            currentOperatorChoice.parent()
+                .addClass('AknDropdown-menuLink--active')
+                .addClass('active');
+
+            this.$el.find('.operator .AknActionButton-highlight').html(currentOperatorChoice.text());
+        },
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/choice-filter.js]: Foutre au parent

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 6]
```diff
@@ -118,11 +125,19 @@ function(
         /**
          * @param {Event} e
          */
-        changeFilterType: function (e) {
-            var select = this.$el.find(e.currentTarget);
-            var selectedValue = select.val();
-
-            this._displayFilterType(selectedValue);
+        _onSelectOperator: function (e) {
+            this.$el.find('.AknDropdown-menuLink')
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: highlight

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/metric-filter.js [hunk 7]
```diff
@@ -40,35 +65,62 @@ define(
...
                 return this;
             },
 
             /**
+             * Highlights the current unit
+             *
+             * @param unit
+             */
+            _highlightUnit(unit) {
+                this.$el.find('.unit .AknDropdown-menuLink')
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/metric-filter.js]: mutualize

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/none-filter.js [hunk 8]
```diff
@@ -20,13 +20,13 @@ function($, _, __, AbstractFilter) {
          * @property
          */
         template: _.template(
-            '<button type="button" class="btn filter-criteria-selector oro-drop-opener oro-dropdown-toggle">' +
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/none-filter.js]: TRACHE

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/price-filter.js [hunk 9]
```diff
@@ -15,54 +28,102 @@ define(
...
 
                 return this;
             },
 
             /**
+             * Highlights the current currency
+             *
+             * @param currency
+             */
+            _highlightCurrency(currency) {
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/price-filter.js]: DROPE

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/price-filter.js [hunk 10]
```diff
@@ -76,82 +137,14 @@ define(
                 if (!value.value) {
                     return this.placeholder;
                 } else {
-                    var option = this._getChoiceOption(value.type);
+                    const option = this._getChoiceOption(value.type);
                     return option.label + ' ' + value.value + ' ' + value.currency;
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/price-filter.js]: ligne vide

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-choice-filter.js [hunk 11]
```diff
@@ -47,33 +53,45 @@ define(
...
-            _disableInput: function() {
-                this.$(this.criteriaValueSelectors.value).val('').select2('destroy');
-                this.$(this.criteriaValueSelectors.value).hide();
+            /**
+             * Highlights the current operator
+             *
+             * @param operator
+             */
+            _highlightOperator(operator) {
+                this.$el.find('.operator .AknDropdown-menuLink')
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-choice-filter.js]: MERE

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-rest-choice-filter.js [hunk 12]
```diff
@@ -40,31 +46,43 @@ define(
...
-                this.$(this.criteriaValueSelectors.value).select2(this._getSelect2Config());
-                this.$(this.criteriaValueSelectors.value).show();
-            },
+            /**
+             * Highlights the current operator
+             *
+             * @param operator
+             */
+            _highlightOperator(operator) {
+                this.$el.find('.operator .AknDropdown-menuLink')
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-rest-choice-filter.js]: FUMIER

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/text-filter.js [hunk 13]
```diff
@@ -15,36 +27,27 @@ function($, _, __, AbstractFilter) {
...
                 '<% if (showLabel) { %>' +
                     '<span class="AknFilterBox-filterLabel"><%= label %></span>' +
                 '<% } %>' +
                 '<span class="AknFilterBox-filterCriteria filter-criteria-hint"><%= criteriaHint %></span>' +
                 '<span class="AknFilterBox-filterCaret"></span>' +
             '</div>' +
-            '<% if (canDisable) { %><a href="<%= nullLink %>" class="AknFilterBox-disableFilter disable-filter"><i class="icon-remove hide-text"><%- _.__("Close") %></i></a><% } %>' +
-            '<div class="filter-criteria dropdown-menu" />'
+            '<div class="filter-criteria dropdown-menu" />' +
+            '<% if (canDisable) { %><a href="<%= nullLink %>" class="AknFilterBox-disableFilter disable-filter"><i class="icon-remove hide-text"><%- _.__("Close") %></i></a><% } %>'
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/text-filter.js]: nullink

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js [hunk 14]
```diff
@@ -179,6 +179,13 @@ define(
         },
 
         /**
+         * Closes the panel when the user clicks on Done button
+         */
+        _onDone() {
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: Close

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js [hunk 15]
```diff
@@ -326,13 +333,32 @@ define(
...
+                    beforeclose: $.proxy(function() {
+                        if (this.selectWidget.getWidget().position().left <= this._getLeftStartPosition()) {
+                            return true;
+                        }
+
+                        this.selectWidget.getWidget().css({ left: this._getLeftEndPosition() + 'px' });
+                        this.selectWidget.getWidget().css({ left: this._getLeftStartPosition() + 'px' });
+                        setTimeout(() => this.selectWidget.multiselect('close'), 500);
+
+                        return false;
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: fataro

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js [hunk 16]
```diff
@@ -341,20 +367,26 @@ define(
             this.selectWidget.getWidget().addClass('pimmultiselect');
 
             this.$('.filter-list span:first').replaceWith(
-                '<a id="add-filter-button" href="javascript:void(0);"></a>'
+                '<div id="add-filter-button" >Filters</div>'
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: endure

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js [hunk 17]
```diff
@@ -341,20 +367,26 @@ define(
...
+                    '<div class="AknButton AknButton--apply done-button">'
+                    + __('pim.grid.category_filter.done') +
+                    '</div>'
+                );
+                button.on('click', () => this._onDone());
+
+                const container = $('<div class="AknColumn-bottomButtonContainer"></div>');
+                container.append(button);
+                this.selectWidget.getWidget().append(container);
+            }
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: maitre dent 2 tamplate

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js [hunk 18]
```diff
@@ -123,17 +106,13 @@ function($, _, mediator) {
...
-
-                var longestElement = _.max(elements, function (element) {
-                    var htmlContent = $(element).find('span:first').html();
-                    var length = htmlContent ? htmlContent.length : 0;
-
-                    return length;
-                });
+                const margin = 100;
+                const elements = this.getWidget().find('.ui-multiselect-checkboxes li');
+                const longest = _.max(_.map(elements, function (element) {
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js]: ES6

File: src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+                    <% }); %>
+                </div>
+            </div>
+        </div>
+        <input class="name_input" type="hidden" name="currency_currency" value="<%- selectedCurrency %>"/>
+    </div>
+    <div class="AknFilterChoice-button">
+        <button type="button" class="AknButton AknButton--apply filter-update"><%- updateLabel %></button>
+    </div>
+</div>
```
Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html]: is empty EUR mal plasé

Comment (pierallard) [inline src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html]: [object oejbevet]

File: src/Pim/Bundle/UIBundle/Resources/public/less/components/grid/FilterDate.less [hunk 20]
```diff
@@ -1,5 +1,6 @@
 @import '../../base/variables';
 
+/* @unused */
 .AknFilterDate {
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/components/grid/FilterDate.less]: ARACH

File: src/Pim/Bundle/UIBundle/Resources/public/less/lib/jquery.multiselect.less [hunk 21]
```diff
@@ -181,3 +182,107 @@
...
+        color: @AknDarkBlue !important;
+        padding-left: 0;
+      }
+
+      .ui-helper-reset li {
+        height: 30px;
+      }
+    }
+
+    .ui-multiselect-checkboxes
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/lib/jquery.multiselect.less]: keskekskséca

File: src/Pim/Bundle/UIBundle/Resources/public/lib/bootstrap/js/bootstrap.js [hunk 22]
```diff
@@ -689,6 +689,21 @@
           // if mobile we we use a backdrop because click events don't delegate
           $('<div class="dropdown-backdrop"/>').insertBefore($(this)).on('click', clearMenus)
         }
+
+        const gap = 30;
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/lib/bootstrap/js/bootstrap.js]: Comment

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js (hunk 18 above)
Comment (pierallard) [reply src/Pim/Bundle/DataGridBundle/Resources/public/js/multiselect-decorator.js]: @juliensnz Excuse me, but I didn't succeed transform this into ES6 functions :/

File: src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html (hunk 19 above)
Comment (pierallard) [reply src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html]: done

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js (hunk 15 above)
Comment (pierallard) [reply src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: done

File: src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html (hunk 19 above)
Comment (pierallard) [reply src/Pim/Bundle/DataGridBundle/Resources/public/templates/filter/price-filter.html]: done

**Comment by nidup** [comment]
👍

**Pull request merged**